micro-rdk/src/common/ota.rs

```diff
@@ -364,14 +386,20 @@ impl<S: OtaMetadataStorage> OtaService<S> {
         let mut total_downloaded: usize = 0;
         let mut got_info = false;
 
+        log::info!("writing new firmware to address `{:#x}`", self.address,);
+
         while let Some(next) = response.frame().await {
-            let frame = next.unwrap();
+            // hyper::Error(Body, Error { kind: Io(Custom { kind: Other, error: "ssl other error -26624" })
+            let frame = next.map_err(|e| OtaError::Other(e.to_string()))?;
             if !frame.is_data() {
                 return Err(OtaError::Other(
                     "download contained non-data frame".to_string(),
                 ));
             }
-            let data = frame.into_data().unwrap();
+            // Err variant returns the original frame, not an impl Error
+            let data = frame
+                .into_data()
+                .map_err(|_| OtaError::Other("".to_string()))?;
```

Comment:
Is there nothing more to say in the error than the emtpy string?

Reply:
updated